include/llbuild/Core/Clock.h

```diff
@@ -0,0 +1,39 @@
+//===- Clock.h --------------------------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLBUILD_CORE_CLOCK_h
+#define LLBUILD_CORE_CLOCK_h
+
+#include <cstdint>
+
+namespace llbuild {
+namespace core {
+
+class Clock {
+public:
+  /// A timestamp is the number of microseconds since a reference date
+  typedef uint64_t Timestamp;
```

Comment:
I would prefer we use Double, and make it seconds, unless there is a compelling reason not to. I always probably wouldn't typedef it, unless we are going to make a wrapper struct to enforce type safety. Defer to @dmbryson on this though.

Reply:
Agreed. I think storing it as a double makes sense.

Reply:
I changed it to double and seconds. My intention was that double is potentially not precise enough and storage. But uint64 uses 64bits as well so that doesn't matter and precision should be at least the same (if not better) as having microsecond integer.

Reply:
Actually - I didn't change that specific part to be double but the representation in the database. This timestamp (in Clock.h) is now a specific time instance which should only be used to calculate durations.

Reply:
Timestamp will be a double and means the number of seconds sind the clock's epoch time.